providers/amazon/src/airflow/providers/amazon/aws/hooks/mwaa.py

```diff
@@ -56,30 +64,78 @@ def invoke_rest_api(
 
         :param env_name: name of the MWAA environment
         :param path: Apache Airflow REST API endpoint path to be called
-        :param method: HTTP method used for making Airflow REST API calls
+        :param method: HTTP method used for making Airflow REST API calls: 'GET'|'PUT'|'POST'|'PATCH'|'DELETE'
         :param body: Request body for the Apache Airflow REST API call
         :param query_params: Query parameters to be included in the Apache Airflow REST API call
+        :param only_use_web_login: If True, only the web login method is used without trying boto's
```

Comment:
nit: maybe just `generate_local_token` since a token is being created either way? It's just who/where it's done?

Reply:
How about `only_generate_local_token` or something that doesn't imply a local token wouldn't be generated on failure in the other case as well?

Reply:
I'm not sure the `only` is required here, honestly. I think if you say `generate_local_token=True` the user expectation is that it will do just that. And you can expand on that in the param description as you are. But also I'm happy to disagree and commit on this one! :smiley: